### PR TITLE
fix install method of cereal

### PR DESCRIPTION
**Pull request by healther**
The include line failed when I tried to install it, I'm not too sure why it is there in the first place, removing it solved my problem and it seems to be a working installation.

### CODE REVIEW

**Comment by adamjstewart** [comment]
What part failed? Do these directories not exist in the build directory? What about the source directory (if you remove the `with working_dir()` stuff)?

I think I messed this up in #4975. @eschnett seems to use this package a lot, can you comment?

**Comment by healther** [comment]
The original error message:
```
$ spack install cereal % gcc@4.9.2                                                                                                                                                            
==> cmake is already installed in ***/opt/spack/linux-debian8-x86_64/gcc-4.9.2/cmake-3.9.0-gsazx53v6h5ywmf6kg4a4en7nt6wst3t
==> Installing cereal
==> Using cached archive: ***/var/spack/cache/cereal/cereal-1.2.2.tar.gz
==> Staging archive: ***/var/spack/stage/cereal-1.2.2-ym5dbr4w3ub7cbdcfn3wql6fls5fn6uh/v1.2.2.tar.gz
==> Created stage in ***/var/spack/stage/cereal-1.2.2-ym5dbr4w3ub7cbdcfn3wql6fls5fn6uh
==> Applied patch Boost2.patch
==> Applied patch Boost.patch
==> Applied patch pointers.patch
==> Ran patch() for cereal
==> Building cereal [CMakePackage]
==> Executing phase : 'cmake'
==> Executing phase : 'build'
==> Executing phase : 'install'
==> Error: OSError: [Errno 2] No such file or directory: 'include'
***/var/spack/repos/builtin/packages/cereal/package.py:69, in install:
     66       def install(self, spec, prefix):
     67           with working_dir(self.build_directory):
     68               install_tree('doc', prefix.doc)
  >> 69               install_tree('include', prefix.include)

See build log for details:
  ***/spack-stage/spack-stage-QyHa_U/cereal-1.2.2/spack-build.out
```

with the latest commit it builds again without error message, but that might not necessarily mean that it works correctly, as my previous deletion also resulted in a fail-free install...

**Comment by davydden** [comment]
>  but that might not necessarily mean that it works correctly, as my previous deletion also resulted in a fail-free install

you can simply check if you have things installed in `prefix.include`, right? And if so, then this PR can be closed.

**Comment by healther** [comment]
Done and this looks good to me (but I haven't used cereal myself)
```
ls opt/spack/linux-debian8-x86_64/gcc-4.9.2/cereal-1.2.2-ym5dbr4w3ub7cbdcfn3wql6fls5fn6uh/include/cereal/
access.hpp  archives  cereal.hpp  details  external  macros.hpp  types
```

**Comment by davydden** [comment]
ok, let's close it unless someone finds a problem later.

**Comment by healther** [comment]
No sorry, it definitely fails with the current ```HEAD``` and what I posted above is the result of a build with the current state of this PR

**Comment by adamjstewart** [comment]
Let's wait for @eschnett to comment. He wrote the original install method, and should know why it was designed the way it previously was.

**Comment by healther** [comment]
@eschnett did you had the chance to look over this?

**Comment by healther** [comment]
@scheibelp @eschnett could you take a look at this one too?

File: var/spack/repos/builtin/packages/cereal/package.py [hunk 1]
```diff
@@ -66,4 +66,4 @@ def cmake_args(self):
     def install(self, spec, prefix):
         with working_dir(self.build_directory):
             install_tree('doc', prefix.doc)
-            install_tree('include', prefix.include)
+        install_tree('include', prefix.include)
```
Comment (junghans) [inline var/spack/repos/builtin/packages/cereal/package.py]: Would `-DJUST_INSTALL_CEREAL=On` work, too?

**Comment by junghans** [comment]
I fixed the `Werror` issue in USCiLab/cereal#337, so adding `-DWITH_WERROR=OFF` should allow us to drop `patch()`.

File: var/spack/repos/builtin/packages/cereal/package.py (hunk 1 above)
Comment (healther) [reply var/spack/repos/builtin/packages/cereal/package.py]: Seems to work

**Comment by healther** [comment]
I took the liberty to remove the patch function too, builds without complaining on my frontend

**Pull request merged**